src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderHelper.java

```diff
@@ -364,12 +365,32 @@ <T> CtFieldAccess<T> createFieldAccess(SingleNameReference singleNameReference)
 			if (ref.isStatic() && !ref.getDeclaringType().isAnonymous()) {
 				va.setTarget(jdtTreeBuilder.getFactory().Code().createTypeAccess(ref.getDeclaringType(), true));
 			} else if (!ref.isStatic()) {
-				va.setTarget(jdtTreeBuilder.getFactory().Code().createThisAccess(jdtTreeBuilder.getReferencesBuilder().getTypeReference(singleNameReference.actualReceiverType), true));
+				CtTypeReference<?> owningType = jdtTreeBuilder.getReferencesBuilder().getTypeReference(singleNameReference.actualReceiverType);
+				if (enclosingClassHasFieldWithName(singleNameReference)) {
```

Comment:
It seems like `singleNameReference.actualReceiverType == singleNameReference.fieldBinding().declaringClass` does the job too, however I'm wondering if actualReceiverType is what we want here for the owning type?

Reply:
@I-Al-Istannen can you resolve this?

Reply:
@I-Al-Istannen ping